### ISSUE TITLE
[Help/Info Req.] Hi all - just some noob questions

### ISSUE THREAD

**Issue by ahungry** [EQEmu/Server#3603]
I've recently been poking around the EQEmu server code and had some questions (and an issue or two) I hope someone has the time to answer (or point me to more info).

I was trying to build for GNU/Linux (using Arch) - my lua libs are 5.4 - I updated one file with a check (luabind/src/pcall.cpp) to match the new function signature in lua >= 5.4 for lua_resume, but subsequently ran into errors in lua_bit.cpp (and note that the file appears copied from a site which stopped updating it in 2012).

I assume to build correctly, I need to explicitly install lua 5.2 libs and point the cmake files to them in some way? 

Which leads me to my next thing - for fun, I felt like trying to pull off a little piece and re-write in a language I'm more familiar with - I thought "loginserver" seemed sensible, as I assume this part of the project could be externalized from the rest of the project.

I think I understand the gist of how this works, but the EQPacket code is a bit hard to follow along and understand the data I should be sending back (I have my rewrite listening on UDP port 6000, but I can't yet figure out the correct data to send back when the client starts pinging it, so the client always ends up in a "failed to connect to login server" state).

Are there any docs outside of the code (or someone capable of explaining) the data format/opcodes for the loginserver portion?

It looks like the flow code wise is roughly:

- A client hits the Client Server
- The client server somehow acknowledges / responds (Op_SessionReady? - I can't figure out what else to send - checking with wireshark, it seems to send ~12 bytes with a 0x02 byte in the first position)
- If I could send the proper response back - it seems some type of username/userpassword auth would occur - at which point, I check credentials against the db
- If good, the client server allows the user to pick from the server list, and connect to a world server (here is where the program hands off credentials to run on someone else's game server I think?)

So... TLDR: does anyone have a breakdown of the loginserver packet communications between client/server?  (UML diagrams would be awesome, or even just _where_ to read through in eq_packet to see the outbound data)

I saw some stuff re: Verant DES key in there - are all communications encrypted with it, or is that just used for the password?

TIA!

**Comment by joligario** [comment]
I recommend creating a thread on Discord for discussion. Might get some good feedback.
